### PR TITLE
Inst finish cleaning

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: test/inst_finish_test.rb [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      expect(subject.main).to eq :abort
+    end
+
+    it "returns :auto if going back in installation" do
+      expect(Yast::GetInstArgs).to receive(:going_back).and_return(true)
+
+      expect(subject.main).to eq :auto
+    end
+
+    it "disabled during its run Back button" do
```
Comment (imobachgs) [inline test/inst_finish_test.rb]: What about *disables Back button while running*?

File: test/inst_finish_test.rb [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      expect(Yast::Wizard).to receive(:DisableBackButton)
+
+      subject.main
+    end
+
+    it "disabled during its run Next button" do
```
Comment (imobachgs) [inline test/inst_finish_test.rb]: Same as above.

File: test/inst_finish_test.rb [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+      it "opens dialog unless already opened" do
+        allow(Yast::SlideShow).to receive(:HaveSlideWidget).and_return(false)
+
+        expect(Yast::SlideShow).to receive(:OpenDialog)
+
+        subject.main
+      end
+
+      it "closes dialog if it is opened by method" do
```
Comment (imobachgs) [inline test/inst_finish_test.rb]: np: I think that the adjective is *open* and the verb past form is *opened*.

File: test/inst_finish_test.rb [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    end
+
+    context "finish clients" do
+      it "Call info for all specified finish clients" do
+        expect(Yast::WFM).to receive(:CallFunction).at_least(1).and_return({})
+
+        subject.main
+      end
+
+      it "filter out all cliets which 'when' key do not match current mode" do
```
Comment (imobachgs) [inline test/inst_finish_test.rb]: cliets -> clients

File: test/inst_finish_test.rb (hunk 1 above)
Comment (jreidinger) [inline test/inst_finish_test.rb]: it is actually not while running as it keep disabled. I won't re enable it or restore previous status.

Comment (jreidinger) [inline test/inst_finish_test.rb]: but you are right, I want to write disables..so maybe simple `disables Back button` ?

File: test/inst_finish_test.rb (hunk 3 above)
Comment (jreidinger) [inline test/inst_finish_test.rb]: I plan to use passive form, not adjective for it.  http://www.englishpage.com/verbpage/activepassive.html so it should be past form here. @mvidner your opinion?

Comment (jreidinger) [inline test/inst_finish_test.rb]: so idea is `method opens dialog` and `dialog is opened by method`

Comment (imobachgs) [inline test/inst_finish_test.rb]: You're right.

**Comment by imobachgs** [comment]
Nice improvement, thanks. Now it looks like Ruby code. LGTM.

File: package/yast2-installation.changes [hunk 5]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Thu Jul 21 11:52:59 UTC 2016 - jreidinger@suse.com
+
+- simplify and speed up inst_finish client(bnc#986649)
```
Comment (imobachgs) [inline package/yast2-installation.changes]: Just add an space before the `bnc`.

**Comment by imobachgs** [comment]
LGTM

**Pull request merged**